pkg/operator/rules.go

```diff
@@ -73,6 +73,29 @@ func NewPrometheusRuleSelector(ruleFormat RuleConfigurationFormat, version strin
 func (prs *PrometheusRuleSelector) generateRulesConfiguration(promRule *monitoringv1.PrometheusRule) (string, error) {
 	logger := log.With(prs.logger, "prometheusrule", promRule.Name, "prometheusrule-namespace", promRule.Namespace)
 	promRuleSpec := promRule.Spec
+
+	promRuleSpec = prs.validateRuleFieldsWithVersion(promRuleSpec)
+
+	content, err := yaml.Marshal(promRuleSpec)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to marshal content")
+	}
+
+	errs := ValidateRule(promRuleSpec)
+	if len(errs) != 0 {
+		const m = "Invalid rule"
+		level.Debug(logger).Log("msg", m, "content", content)
+		for _, err := range errs {
+			level.Info(logger).Log("msg", m, "err", err)
+		}
+		return "", errors.New(m)
+	}
+	return string(content), nil
+}
+
+// validateRuleFieldsWithVersion checks whether fields added in Rule object is supported in respective Prometheus/Thanos versions.
+func (prs *PrometheusRuleSelector) validateRuleFieldsWithVersion(promRuleSpec monitoringv1.PrometheusRuleSpec) monitoringv1.PrometheusRuleSpec {
```

Comment:
(nit) it's more about sanitizing the spec than validating it.
```suggestion
func (prs *PrometheusRuleSelector) sanitizePrometheusRulesSpec(promRuleSpec monitoringv1.PrometheusRuleSpec) monitoringv1.PrometheusRuleSpec {
```